deploy/ui/form.json

```diff
@@ -107,26 +107,78 @@
+          "label": "Admin user configuration",
+          "type": "dropDown",
           "required": false,
-          "jsonPath": "$.spec.commonConfig",
           "visible": true,
+          "default": "default",
+          "options": [
+            {
+              "value": "secret",
+              "label": "Secret configuration"
+            },
+            {
+              "value": "default",
+              "label": "Username and Password configuration"
+            }
+          ],
           "fields": [
             {
-              "label": "Username",
-              "default": "adminUser",
+              "label": "Username and password",
+              "type": "fieldGroup",
               "required": false,
-              "description": "Administrative username for RHPAM components",
-              "jsonPath": "$.spec.commonConfig.adminUser",
-              "type": "text"
+              "jsonPath": "$.spec.commonConfig",
+              "displayWhen": "default",
+              "visible": true,
+              "fields": [
+                {
+                  "label": "Username",
+                  "default": "adminUser",
+                  "required": false,
+                  "description": "Administrative username for RHPAM components",
+                  "jsonPath": "$.spec.commonConfig.adminUser",
+                  "type": "text"
+                },
+                {
+                  "label": "Password",
+                  "required": false,
+                  "description": "Administrative password, if empty will be generated",
+                  "jsonPath": "$.spec.commonConfig.adminPassword",
+                  "type": "password"
+                }
+              ]
             },
             {
-              "label": "Password",
+              "label": "OpenShift admin credentials secret",
+              "type": "fieldGroup",
               "required": false,
-              "description": "Administrative password to use instead of generating default value",
-              "jsonPath": "$.spec.commonConfig.adminPassword",
-              "type": "password"
+              "jsonPath": "$.spec.commonConfig.secretAdminCredentials",
+              "displayWhen": "secret",
+              "visible": false,
+              "fields": [
+                {
+                  "label": "Secret that holds admin credentials information",
+                  "required": false,
+                  "description": "If the provided secret is not found or is missing, one with the name kie-admin-credentials secret will be created with the given username and password.",
+                  "jsonPath": "$.spec.commonConfig.secretAdminCredentials",
+                  "type": "text"
+                },
+                {
+                  "label": "Username",
+                  "default": "adminUser",
+                  "required": false,
+                  "description": "Administrative user, if leaved empty adminUser will be used",
```

Comment:
needs better description here. Not clear that it is not needed when user is already available in secret. It should be also marked as optional property.